controller/volume_controller.go

```diff
@@ -2999,6 +3000,127 @@ func (vc *VolumeController) ResolveRefAndEnqueue(namespace string, ref *metav1.O
 	vc.enqueueVolume(volume)
 }
 
+// createNewRecurringJobNameUsed use a new name to create a new recurring job that original name exists but configuration is different
+func createNewRecurringJobNameUsed(log *logrus.Entry, ds *datastore.DataStore, job *longhorn.RecordVolumeRecurringJob) (string, error) {
+	newJobName := util.RandomID()
+	if _, err := ds.GetRecurringJob(newJobName); err != nil {
```

Comment:
Do we need to consider if the recurring job exists?

Reply:
Removed. Yeah, it seems that it does not need to check this, just create and return the error if it exists, and extend the re-create recurring job name to 16 chars with a prefix.